api/v1alpha1/paas_types.go

```diff
@@ -257,17 +259,50 @@ type PaasCapability struct {
 	GitRevision string `json:"gitRevision,omitempty"`
 	// the path in the git repo that contains the Applications / Application Sets to be used by this ArgoCD
 	GitPath string `json:"gitPath,omitempty"`
+	// Custom fields to configure this specific Capability
+	CustomFields map[string]string `json:"custom_fields,omitempty"`
 	// This project has it's own ClusterResourceQuota settings
 	Quota paas_quota.Quotas `json:"quota,omitempty"`
 	// You can add ssh keys (which is a type of secret) for ArgoCD to use for access to bitBucket
 	// They must be encrypted with the public key corresponding to the private key deployed together with the PaaS operator
 	SshSecrets map[string]string `json:"sshSecrets,omitempty"`
-	// You can enable extra permissions for the service accounts beloning to this capability
+	// You can enable extra permissions for the service accounts belonging to this capability
 	// Exact definitions is configured in Paas Configmap
 	// Note that we want to remove (some of) these permissions in future releases (like self-provisioner)
 	ExtraPermissions bool `json:"extra_permissions,omitempty"`
 }
 
+func (pc *PaasCapability) CapExtraFields(fieldConfig map[string]config.ConfigCustomField) (fields map[string]string, issues []error) {
+	// TODO: remove argocd specific fields
+	fields = map[string]string{
+		"git_url":      pc.GitUrl,
+		"git_revision": pc.GitRevision,
+		"git_path":     pc.GitPath,
+	}
+	for key, value := range pc.CustomFields {
+		fields[key] = value
+	}
+	for key, fieldConf := range fieldConfig {
+		if value, exists := fields[key]; exists {
+			validation := ".*"
```

Comment:
Can you add this to the paasTypes description as the default validation being applied.

Reply:
Done. And added more docs to config. And added TODO for when moving to PaasConfig.